### ISSUE TITLE
Extraction of 5GB (ZIP64) archive created with Go failed

### ISSUE THREAD

**Issue by philip-firstorder** [MacPaw/XADMaster#100]
Extraction of 5GB (ZIP64) archive created with [Go Archiver](https://golang.org/src/archive/zip/writer.go):  failed

**The Unarchiver (4.1.0):** There was a problem while reading the contents of "Go.zip": Data is corrupted => clicked Continue => Archive 5GB.zip is extracted, but not Small image.png
<img width="458" alt="Screenshot 2019-07-10 at 15 32 04" src="https://user-images.githubusercontent.com/30979663/60973167-04705200-a328-11e9-8dc0-e927f2903903.png">

**The Archive Browser (1.11.2)**
<img width="430" alt="Screenshot 2019-07-10 at 15 31 39" src="https://user-images.githubusercontent.com/30979663/60973119-e60a5680-a327-11e9-8670-8dc461511441.png">

=> Pressed OK and then this is what I see
<img width="792" alt="Screenshot 2019-07-10 at 15 35 43" src="https://user-images.githubusercontent.com/30979663/60973355-6761e900-a328-11e9-92dc-6e8beaaaa929.png">

I archived 2 blank files with ZIP Method: Storage
```
$ mkfile -n 5000000000 ~/Desktop/5GB_big_empty_file
$ mkfile -n 5000 ~/Desktop/5KB_small_empty_file
```

To reproduce please extract [Go.zip.zip](https://github.com/MacPaw/XADMaster/files/3377689/Go.zip.zip)  (5,9 MB) once to get **Go.zip (5 GB)** => then try to extract it with The Unarchiver (4.1.0) or open with The Archive Browser (1.11.2)

And [here is the explanation](https://github.com/aonez/Keka/issues/423#issuecomment-508703154) on what went wrong from original issue #https://github.com/aonez/Keka/issues/423

@PaulTaykalo 
@aonez Knows the technical details and fixed it for Keka.

**Comment by philip-firstorder** [comment]
#https://github.com/aonez/Keka/issues/423

**Comment by philip-firstorder** [comment]
Sure, here are some big 5GB [sample archives](https://github.com/aonez/Keka/issues/423#issuecomment-507613049) created with Go, Keka and MacOS.

This is also the current detailed [unarchiving status](https://github.com/aonez/Keka/issues/423#issuecomment-507613049).

I already tested a release candidate of [Keka unar (1.1.18)](https://github.com/aonez/Keka/releases/download/dev-test-builds/Keka-Go64.7z) with this issue fixed.

**Comment by PaulTaykalo** [comment]
Well, it is more like this part is related to the fact that these changes are "broken"
https://github.com/MacPaw/XADMaster/blob/master/XADZipParser.m#L329
```
				off_t uncompsize64=[fh readUInt64LE];
				off_t compsize64=[fh readUInt64LE];
				off_t locheaderoffset64=[fh readUInt64LE];
				int startdisk64=[fh readUInt32LE]; <<<< (This is kinda optional, depending on the flags)
				if(uncompsize==0xffffffff) uncompsize=uncompsize64;
```

Will make a fix bit later. With this line commented out, unarchiving works fine

**Comment by philip-firstorder** [comment]
Yes, that's it, go and other archivers don't define the **startdisk** at all when it's not informed with 0xffff, this because it's also the **last** field so it can be safely ignored.

**Comment by aonez** [comment]
Already did the pull: https://github.com/MacPaw/XADMaster/pull/101

I'm using this fixed code for half a year now, it's safe.

**Comment by PaulTaykalo** [comment]
Fixed in 8662695
